mlflow/tracking/__init__.py

```diff
@@ -201,39 +199,41 @@ def _get_unix_timestamp():
     return int(time.time() * 1000)
 
 
-def _do_start_run(run_uuid=None, experiment_id=None, source_name=None, source_version=None,
-                  entry_point_name=None, source_type=None):
+def _create_run(experiment_id, source_name, source_version, entry_point_name, source_type):
     store = _get_store()
-
-    if run_uuid is not None:
-        existing_run = store.get_run(run_uuid)
-        if existing_run is None:
-            raise Exception("Could not start run with UUID %s - no such run found." % run_uuid)
-        updated_info = store.update_run_info(
-            run_uuid=run_uuid, run_status=RunStatus.RUNNING, end_time=None)
-        return ActiveRun(updated_info, store)
-
-    # Get experiment ID for run
-    exp_id_for_run = experiment_id or _get_experiment_id()
-    run = store.create_run(experiment_id=exp_id_for_run, user_id=_get_user_id(), run_name=None,
-                           source_type=(source_type or _get_source_type()),
-                           source_name=(source_name or _get_source_name()),
+    run = store.create_run(experiment_id=experiment_id, user_id=_get_user_id(), run_name=None,
+                           source_type=source_type,
+                           source_name=source_name,
                            entry_point_name=entry_point_name,
                            start_time=_get_unix_timestamp(),
-                           source_version=(source_version or _get_source_version()), tags=[])
+                           source_version=source_version, tags=[])
     return ActiveRun(run.info, store)
 
 
+def get_run(run_uuid):
```

Comment:
Maybe we should also allow for passing a `tracking_uri` to get run information from a remote tracking server (e.g. Databricks)?

Reply:
I think we'd have to put that on all methods to be consistent, so I prefer to require people to do `set_tracking_uri` right now.